From 2a1a89a6e94b90f2feea2b1a285330eaa2707c0c Mon Sep 17 00:00:00 2001
From: Gabriel de Araújo Cabral <gabriel.cabral@windriver.com>
Date: Thu, 4 Apr 2024 16:10:05 -0300
Subject: [PATCH] ceph-csi-cephfs: add annotations to
 provisioner-deployment.yaml

This patch adds annotations to the cephfs deployment so that
disabling the chart works correctly.

Signed-off-by: Gabriel de Araújo Cabral <gabriel.cabral@windriver.com>
---
 charts/ceph-csi-cephfs/templates/provisioner-deployment.yaml | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/charts/ceph-csi-cephfs/templates/provisioner-deployment.yaml b/charts/ceph-csi-cephfs/templates/provisioner-deployment.yaml
index 4767866ba..d2ae00510 100644
--- a/charts/ceph-csi-cephfs/templates/provisioner-deployment.yaml
+++ b/charts/ceph-csi-cephfs/templates/provisioner-deployment.yaml
@@ -10,6 +10,9 @@ metadata:
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
     {{- with .Values.commonLabels }}{{ toYaml . | trim | nindent 4 }}{{- end }}
+  annotations:
+    "helm.sh/hook": "post-upgrade, post-install, post-delete"
+    "helm.sh/hook-delete-policy": "before-hook-creation"
 spec:
   replicas: {{ .Values.provisioner.replicaCount }}
   strategy:
@@ -147,7 +150,7 @@ spec:
             - "--v={{ .Values.sidecarLogLevel }}"
             - "--timeout={{ .Values.provisioner.timeout }}"
             - "--leader-election=true"
-            - "--extra-create-metadata=true"
+            - "--extra-create-metadata=false"
             - "--enable-volume-group-snapshots={{.Values.provisioner.snapshotter.args.enableVolumeGroupSnapshots }}"
 {{- range .Values.provisioner.snapshotter.extraArgs }}
             - "--{{ . }}"
--
2.25.1
